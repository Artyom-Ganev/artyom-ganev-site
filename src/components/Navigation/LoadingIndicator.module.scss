.container {
  background-image: url("../../resources/loading-indicator.svg");
  width: 120px;
  height: 120px;
}
