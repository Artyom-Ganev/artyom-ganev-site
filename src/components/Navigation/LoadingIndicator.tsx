import React, { FunctionComponent } from 'react';

import { loadingIndicator } from 'assets';

import styles from './LoadingIndicator.module.scss';

/**
 * Simple loading indicator based on SVG-image
 */
const LoadingIndicator: FunctionComponent = () => (
  <div className='flexBox flexGrow-1 flexShrink-1 justifyContentCenter'>
    <div className={styles.container}>{loadingIndicator}</div>
  </div>
);

export default LoadingIndicator;
