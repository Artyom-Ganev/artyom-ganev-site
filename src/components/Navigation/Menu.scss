.nav-menu {
  &__root {
    background-color: #FFFFFF;
    position: fixed;
    width: 100%;
    z-index: 1;
  }

  &__item {
    font-weight: bold;
    padding: 0.8em 1em;
    color: #343648;

    &.active {
      background-color: #ba96ff;
    }

    &:visited {
      color: #343648;
    }

    &:hover {
      background-color: #ba96ff;
      cursor: pointer;
      text-decoration: none;
      color: #FFFFFF;
    }
  }
}