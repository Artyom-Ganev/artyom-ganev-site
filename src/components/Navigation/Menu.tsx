import React, { FunctionComponent, ReactNode } from 'react';
import { NavLink } from 'react-router-dom';

import { PAGES } from 'const';

import styles from './Menu.module.scss';

/**
 * Menu items render function
 */
const renderItems = (): ReactNode[] =>
  PAGES.map(
    ({ id, name, url }): ReactNode => (
      <NavLink exact={!url} className={styles.item} activeClassName={styles.active} to={`/${url}`} key={id}>
        {name}
      </NavLink>
    )
  );

/**
 * /**
 * Navigation menu component
 */
const Menu: FunctionComponent = () => (
  <nav className={`flexBox ${styles.container}`} id='menu'>
    {renderItems()}
  </nav>
);

export default Menu;
