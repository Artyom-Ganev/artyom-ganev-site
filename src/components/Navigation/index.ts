export { default as LoadingIndicator } from './LoadingIndicator';
export { default as NavigationMenu } from './Menu';
export { default as Routes } from './Routes';
