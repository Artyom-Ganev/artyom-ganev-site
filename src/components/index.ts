export { default as App, NarrowContext } from './App';
