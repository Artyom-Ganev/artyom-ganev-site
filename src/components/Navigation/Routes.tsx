import React, { FunctionComponent, lazy, Suspense } from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { LoadingIndicator } from 'src/components/Navigation';
import { PageContainer } from 'src/components/Pages';

const Main = lazy(() => import('src/components/Pages/Main'));
const About = lazy(() => import('src/components/Pages/About'));
const Blog = lazy(() => import('src/components/Pages/Blog'));
const Career = lazy(() => import('src/components/Pages/Career'));
const Contacts = lazy(() => import('src/components/Pages/Contacts'));
const NotFound = lazy(() => import('src/components/Pages/NotFound'));

const Indicator: FunctionComponent = () => (
  <PageContainer>
    <LoadingIndicator />
  </PageContainer>
);

/**
 * Application routes component
 */
const Routes: FunctionComponent = () => (
  <Suspense fallback={<Indicator />}>
    <Switch>
      <Redirect from='/main' to='/' />
      <Route exact={true} path='/' component={Main} />
      <Route exact={true} path='/about' component={About} />
      <Route exact={true} path='/blog' component={Blog} />
      <Route exact={true} path='/career' component={Career} />
      <Route exact={true} path='/contacts' component={Contacts} />
      <Route component={NotFound} />
    </Switch>
  </Suspense>
);

export default Routes;
