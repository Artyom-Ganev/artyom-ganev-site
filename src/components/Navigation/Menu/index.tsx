import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { PopoverOrigin } from '@material-ui/core/Popover/Popover';
import React, { FunctionComponent, MouseEvent, ReactNode, useCallback, useState } from 'react';
import NavigationMenuItem from 'src/components/Navigation/Menu/Item';
import { PAGES } from 'src/const';

export const ANCHOR_ORIGIN: PopoverOrigin = {
  vertical: 'bottom',
  horizontal: 'left',
} as const;

export const TRANSFORM_ORIGIN: PopoverOrigin = {
  vertical: 'top',
  horizontal: 'left',
} as const;

export const MENU_TITLE = 'Open Menu';

/**
 * /**
 * Navigation menu component
 */
const NavigationMenu: FunctionComponent = () => {
  const [anchor, setAnchor] = useState<null | HTMLElement>(null);

  const open = useCallback(
    ({ currentTarget }: MouseEvent<HTMLElement>) => {
      setAnchor(currentTarget);
    },
    [setAnchor]
  );

  const close = useCallback(() => {
    setAnchor(null);
  }, [setAnchor]);

  return (
    <nav>
      <Button variant='contained' color='primary' onClick={open}>
        {MENU_TITLE}
      </Button>
      <Menu
        anchorEl={anchor}
        anchorOrigin={ANCHOR_ORIGIN}
        elevation={0}
        getContentAnchorEl={null}
        id='menu'
        keepMounted
        onClose={close}
        open={Boolean(anchor)}
        transformOrigin={TRANSFORM_ORIGIN}
      >
        {PAGES.map(
          ({ id, name, url }): ReactNode => (
            <MenuItem disableGutters key={id}>
              <NavigationMenuItem name={name} onClick={close} url={url} />
            </MenuItem>
          )
        )}
      </Menu>
    </nav>
  );
};

export default NavigationMenu;
