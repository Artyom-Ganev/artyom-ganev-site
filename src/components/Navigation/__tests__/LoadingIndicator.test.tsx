import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { LoadingIndicator } from 'src/components/Navigation';

describe('LoadingIndicator', () => {
  const component = shallow(<LoadingIndicator />);

  it('Should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
