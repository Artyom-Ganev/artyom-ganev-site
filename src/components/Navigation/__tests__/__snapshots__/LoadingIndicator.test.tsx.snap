// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Should match snapshot 1`] = `
<div
  className="flexBox flexGrow-1 flexShrink-1 justifyContentCenter"
>
  <div
    className="container"
  />
</div>
`;
